src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -429,11 +437,35 @@ namespace HeatBalanceSurfaceManager {
 		SetSurfaceOutBulbTempAt();
 		CheckSurfaceOutBulbTempAt();
 
+		// X Luo Added 07/19/2017
+		// set surface level wind dir to global value
 		SetSurfaceWindSpeedAt();
+		SetSurfaceWindDirAt();
 		//  DO SurfNum = 1, TotSurfaces
 		//    IF (Surface(SurfNum)%ExtWind) Surface(SurfNum)%WindSpeed = WindSpeedAt(Surface(SurfNum)%Centroid%z)
 		//  END DO
 
+		// X Luo Added 07/30/2017
+		// Set surface data to linked air node value 
+		for ( SurfNum = 1; SurfNum <= TotSurfaces; ++SurfNum ) {
+			if ( Surface( SurfNum ).HasLinkedOutAirNode ) {
+				if ( Node( Surface( SurfNum ).LinkedOutAirNode ).SchedOutAirDryBulb ) {
+					Surface( SurfNum ).OutDryBulbTemp = GetCurrentScheduleValue( Node( Surface( SurfNum ).LinkedOutAirNode ).OutAirDryBulbSchedNum );
+				}
+				if ( Node( Surface( SurfNum ).LinkedOutAirNode ).SchedOutAirWetBulb ) {
+					Surface( SurfNum ).OutWetBulbTemp = GetCurrentScheduleValue( Node( Surface( SurfNum ).LinkedOutAirNode ).OutAirWetBulbSchedNum );
+				}
+				if ( Node( Surface( SurfNum ).LinkedOutAirNode ).SchedOutAirWindSpeed ) {
+					Surface( SurfNum ).WindSpeed = GetCurrentScheduleValue( Node( Surface( SurfNum ).LinkedOutAirNode ).OutAirWindSpeedSchedNum );
+				}
+				if ( Node( Surface( SurfNum ).LinkedOutAirNode ).SchedOutAirWindDir ) {
+					Surface( SurfNum ).WindDir = GetCurrentScheduleValue( Node( Surface( SurfNum ).LinkedOutAirNode ).OutAirWindDirSchedNum );
+				}
+			}
+		}
+
+		// X Luo Added 07/19/2017
+		// Overwriting surface and zone level environmental data with EMS override value 
 		if ( AnyEnergyManagementSystemInModel ) {
 			for ( SurfNum = 1; SurfNum <= TotSurfaces; ++SurfNum ) {
 				if ( Surface( SurfNum ).OutDryBulbTempEMSOverrideOn ) {
```

Comment:
We added a global flag `AnyLocalEnvironmentsInModel` in parallel with `AnyEnergyManagementSystemInModel `. These conditions normally don't come at a time. I'd say using the flag and skip either routine won't harm. Don't know if I understand this right.